File: services/administration-service/src/main/java/sgptt/adminsvc/entity/CriterionEvaluation.java
package sgptt.adminsvc.entity;

import java.util.Date;

import jakarta.persistence.*;
import sgptt.adminsvc.model.Criterion;

@Entity
public class CriterionEvaluation {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "evaluation_id")
    private Long evaluationId;

    @Column(name = "criterion_result", nullable = false, columnDefinition = "BOOLEAN DEFAULT FALSE")
    private boolean criterionResult;

    @Enumerated(EnumType.ORDINAL)
    private Criterion criterion;

    @Column(name = "criterion_comments", columnDefinition = "TEXT", nullable = false)
    private String criterionComments;

    @Temporal(TemporalType.TIMESTAMP)
    private Date evaluationDate;

}

File: services/auth-service/src/main/kotlin/com/sgpttt/authservice/repository/entity/CriterionEvaluation.kt
package com.sgpttt.authservice.repository.entity

import com.sgpttt.authservice.model.domain.Criterion
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Temporal
import jakarta.persistence.TemporalType
import java.util.Date

@Entity
data class CriterionEvaluation (
	// Getters and Setters
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "evaluation_id")
	val evaluationId: Long,
	
	@Column(name = "criterion_result", nullable = false, columnDefinition = "BOOLEAN DEFAULT FALSE")
	val criterionResult: Boolean = false,
	
	@Enumerated(EnumType.ORDINAL)
	val criterion: Criterion,
	
	@Column(name = "criterion_comments", columnDefinition = "TEXT", nullable = false)
	val criterionComments: String,
	
	@Temporal(TemporalType.TIMESTAMP)
	val evaluationDate: Date
)